src/sentry/search/utils.py

```diff
@@ -454,19 +466,21 @@ def parse_release(
 ]
 
 
-def get_numeric_field_value(field, raw_value, type=int):
+def get_numeric_field_value(
+    field: str, raw_value: str, type: Type = int
+) -> dict[str, Union[int, float]]:
     try:
         for modifier, function in numeric_modifiers:
             if raw_value.startswith(modifier):
-                return function(field, type(raw_value[len(modifier) :]))
+                return function(field, type(raw_value[len(modifier) :]))  # type: ignore[no-untyped-call]
```

Comment:
The structure of `numeric_modifiers` seems pretty strict, you could type `numeric_modifiers` using a `Callable` for the lambdas.